yunwoo/programmers-0/평행.js

```diff
@@ -0,0 +1,27 @@
+function solution(dots) {
+  const radiansObj = {};
+  let result = 0;
+
+  for (let i = 0; i < dots.length - 1; i++) {
+    const [x, y] = dots[i];
+
+    for (let j = i + 1; j < dots.length; j++) {
+      const [dx, dy] = dots[j];
+      const radian = Math.floor(
+        calcAngleDegrees(Math.abs(x - dx), Math.abs(y - dy))
+      );
+
+      radiansObj[radian] = (radiansObj[radian] || 0) + 1;
+
+      if (radiansObj[radian] >= 2) {
+        result++;
+      }
+    }
+  }
```

Comment:
- 와우~ atan2를 이용해서 각도를 이용한 풀이 잘봤습니다. 
```suggestion
    for (let j = i + 1; j < dots.length; j++) {
      const [nx, ny] = dots[j];
      const radian = Math.floor(
        calcAngleDegrees(Math.abs(x - nx), Math.abs(y - ny))
      );

      radiansObj[radian] = (radiansObj[radian] || 0) + 1;

      if (radiansObj[radian] >= 2) {
        result++;
      }
    }
  }
```
- dx, dy는 보통 내가 가고자하는 방향?을 의미하는 경우 즉 deltaX, deltaY를 의미하기 때문에 여기서는 다음 비교 대상인 x,y를 의미하는 nx, ny 가 적절해보입니다. nextX, nextY
- 아래 코드는 각도가 아닌 기울기 공식을 이용한 풀이입니다 참고해주시길 바랍니다 🙇‍♂️
```js
function getSlope(x1, y1, x2, y2) {
  if (x2 - x1 === 0) return 0;
  return (y2 - y1) / (x2 - x1);
}

function solution(dots) {
  const slopes = [];

  for (let i = 0; i < dots.length; i++) {
    for (let j = i + 1; j < dots.length; j++) {
      const slope = getSlope(dots[i][0], dots[i][1], dots[j][0], dots[j][1]);

      if (slopes.includes(slope)) return 1;
      slopes.push(slope);
    }
  }
  return 0;
}
```

Reply:
- 아하 델타가 방향의 의미를 가졌었군요!! 참고하겠습니다 감사합니다👍
- 참고로 주신 풀이가 값이 나올경우 연산이 바로 멈춰서 효율이 더 좋군요!! 아직 좌표가 들어가면 많이 헷갈리네요🥲 참고풀이 감사합니다!!🙇‍♂️